pkg/apis/autoscaling/v1alpha1/pa_validation_test.go

```diff
@@ -231,6 +269,29 @@ func TestPodAutoscalerValidation(t *testing.T) {
 			},
 		},
 		want: apis.ErrOutOfBoundsValue("FOO", 1, math.MaxInt32, autoscaling.MinScaleAnnotationKey).ViaField("metadata", "annotations"),
+	}, {
+		name: "bad metric",
+		r: &PodAutoscaler{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "valid",
+				Annotations: map[string]string{
+					"autoscaling.knative.dev/metric": "memory",
+				},
+			},
+			Spec: PodAutoscalerSpec{
+				ScaleTargetRef: corev1.ObjectReference{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+					Name:       "bar",
+				},
+				ProtocolType: net.ProtocolH2C,
+			},
+		},
+		want: &apis.FieldError{
+			Message: fmt.Sprintf("Unsupported metric %q for PodAutoscaler class %q",
```

Comment:
Should this be apis.ErrInvalidValue?

Reply:
The original one is FieldError. If ErrInvalidValue makes more sense, I can do it in a separate PR.

Reply:
Yeah, this is definitely invalid value for the annotation.